src/libos/src/entry.rs

```diff
@@ -240,6 +252,32 @@ fn parse_arguments(
     Ok((path_buf, args, env_merged, host_stdio_fds))
 }
 
+fn parse_resolv_conf(resolv_conf_ptr: *const c_char) -> Result<()> {
+    // Read resolv.conf file from host
+    let resolv_conf_bytes = unsafe { CStr::from_ptr(resolv_conf_ptr).to_bytes() };
+    let resolv_conf_str = str::from_utf8(resolv_conf_bytes)
+        .map_err(|_| errno!(EINVAL, "/etc/resolv.conf contains non UTF-8 characters"))?;
+
+    // Parse and inspect resolv.conf file
+    if let Err(_) = resolv_conf::Config::parse(resolv_conf_bytes) {
+        return_errno!(EINVAL, "malformated host /etc/resolv.conf");
+    }
+
+    let fs_view = FsView::new();
```

Comment:
This function not only parse the resolv.conf file but also copy it to a local file. Should be a separate function  to do the copy.

Reply:
Good point.